Author: aszlig <aszlig@nix.build>
Date:   Tue Apr 3 05:09:43 2018 +0200

    Generate HTML test reports as well as stdio output
    
    We want to have nice HTML reports for Hydra instead of just the whole
    text blurb, so let's use mocha-simple-html-reporter for that.
    
    In addition I added the mocha-multi-reporters plugin because we still
    want the text blurb while running tests outside of Hydra.
    
    Signed-off-by: aszlig <aszlig@nix.build>
    Filename: html-test-reports.patch

diff --git a/gulp/gulp-tests.js b/gulp/gulp-tests.js
index d2b451cb0..cf2dfa494 100644
--- a/gulp/gulp-tests.js
+++ b/gulp/gulp-tests.js
@@ -170,7 +170,7 @@ gulp.task('test:content:safe', gulp.series('test:prepare:build', (cb) => {
 
 gulp.task('test:api-v3:unit', (done) => {
   let runner = exec(
-    testBin('istanbul cover --dir coverage/api-v3-unit mocha -- test/api/v3/unit --recursive --require ./test/helpers/start-server'),
+    testBin('istanbul cover --dir coverage/api-v3-unit mocha -- test/api/v3/unit --reporter mocha-multi-reporters --reporter-options configFile=mocha-reporters.json --recursive --require ./test/helpers/start-server'),
     CONTENT_OPTIONS,
     (err) => {
       if (err) {
@@ -189,7 +189,7 @@ gulp.task('test:api-v3:unit:watch', () => {
 
 gulp.task('test:api-v3:integration', (done) => {
   let runner = exec(
-    testBin('istanbul cover --dir coverage/api-v3-integration --report lcovonly mocha -- test/api/v3/integration --recursive --require ./test/helpers/start-server'),
+    testBin('istanbul cover --dir coverage/api-v3-integration --report lcovonly mocha -- test/api/v3/integration --reporter mocha-multi-reporters --reporter-options configFile=mocha-reporters.json --recursive --require ./test/helpers/start-server'),
     CONTENT_OPTIONS,
     (err) => {
       if (err) {
@@ -211,7 +211,7 @@ gulp.task('test:api-v3:integration:watch', () => {
 
 gulp.task('test:api-v3:integration:separate-server', (done) => {
   let runner = exec(
-    testBin('mocha test/api/v3/integration --recursive --require ./test/helpers/start-server', 'LOAD_SERVER=0'),
+    testBin('mocha test/api/v3/integration --reporter mocha-multi-reporters --reporter-options configFile=mocha-reporters.json --recursive --require ./test/helpers/start-server', 'LOAD_SERVER=0'),
     CONTENT_OPTIONS,
     (err) => done(err)
   );
diff --git a/mocha-reporters.json b/mocha-reporters.json
new file mode 100644
index 000000000..347c750ab
--- /dev/null
+++ b/mocha-reporters.json
@@ -0,0 +1,4 @@
+{
+  "reporterEnabled": "spec, mocha-simple-html-reporter",
+  "mochaSimpleHtmlReporterReporterOptions": {"output": "test-report.html"}
+}
diff --git a/package.json b/package.json
index 523692a14..d8450b49b 100644
--- a/package.json
+++ b/package.json
@@ -109,9 +109,9 @@
     "test:api-v3:unit": "gulp test:api-v3:unit",
     "test:api-v3:integration": "gulp test:api-v3:integration",
     "test:api-v3:integration:separate-server": "NODE_ENV=test gulp test:api-v3:integration:separate-server",
-    "test:sanity": "istanbul cover --dir coverage/sanity --report lcovonly mocha -- test/sanity --recursive",
-    "test:common": "istanbul cover --dir coverage/common --report lcovonly mocha -- test/common --recursive",
-    "test:content": "istanbul cover --dir coverage/content --report lcovonly mocha -- test/content --recursive",
+    "test:sanity": "istanbul cover --dir coverage/sanity --report lcovonly mocha -- test/sanity --reporter mocha-multi-reporters --reporter-options configFile=mocha-reporters.json --recursive",
+    "test:common": "istanbul cover --dir coverage/common --report lcovonly mocha -- test/common --reporter mocha-multi-reporters --reporter-options configFile=mocha-reporters.json --recursive",
+    "test:content": "istanbul cover --dir coverage/content --report lcovonly mocha -- test/content --reporter mocha-multi-reporters --reporter-options configFile=mocha-reporters.json --recursive",
     "test:nodemon": "gulp test:nodemon",
     "coverage": "COVERAGE=true mocha --require register-handlers.js --reporter html-cov > coverage.html; open coverage.html",
     "sprites": "gulp sprites:compile",
@@ -160,6 +160,8 @@
     "karma-webpack": "^3.0.0",
     "lcov-result-merger": "^2.0.0",
     "mocha": "^5.1.1",
+    "mocha-multi-reporters": "^1.1.7",
+    "mocha-simple-html-reporter": "^1.1.0",
     "monk": "^6.0.6",
     "nightwatch": "^0.9.21",
     "puppeteer": "^1.4.0",
